repose-aggregator/tests/functional-tests/src/integrationTest/groovy/features/core/powerfilter/ExceptionAndErrorHandlingTest.groovy

```diff
@@ -0,0 +1,60 @@
+/*
+ * _=_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=
+ * Repose
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Copyright (C) 2010 - 2015 Rackspace US, Inc.
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=_
+ */
+package features.core.powerfilter
+
+import org.openrepose.framework.test.ReposeValveTest
+import org.rackspace.deproxy.Deproxy
+import org.rackspace.deproxy.MessageChain
+
+class ExceptionAndErrorHandlingTest extends ReposeValveTest {
+    def setupSpec() {
+        deproxy = new Deproxy()
+        deproxy.addEndpoint(properties.targetPort)
+
+        def params = properties.getDefaultTemplateParams()
+        repose.configurationProvider.applyConfigs("common", params)
+        repose.configurationProvider.applyConfigs("features/core/powerfilter/exceptionanderror", params)
+        repose.start()
+        waitUntilReadyToServiceRequests()
+    }
+
+    def "should return a 500 response if an exception is thrown by a filter during processing"() {
+        when:
+        MessageChain messageChain = deproxy.makeRequest(
+            url: reposeEndpoint + "/exceptionanderror"
+        )
+
+        then:
+        // The PowerFilterChain catches the Exception and sets the response code to 500
+        messageChain.receivedResponse.code.toInteger() == 500
```

Comment:
Magic number, although we do that a lot. the code actually explicitly says we'll throw a 502.